### ISSUE TITLE
`torch.compile` slowing down transformer model

### ISSUE THREAD

**Issue by mattbarrett98** [pytorch/pytorch#104601]
### 🐛 Describe the bug

`torch.compile` is slowing down the execution of `SwinModel` from the transformers library for me on cpu:

```python
from transformers import SwinModel
from time import perf_counter
import torch

model = SwinModel.from_pretrained("microsoft/swin-tiny-patch4-window7-224")
x = torch.rand((1, 3, 224, 224))

s = perf_counter()
model(x)
print("original time", perf_counter() - s)

compiled = torch.compile(model)
s = perf_counter()
compiled(x)
print("compiling time", perf_counter() - s)

s = perf_counter()
compiled(x)
print("compiled model time", perf_counter() - s)
```

### Error logs

```
original time 0.06399844800034771
compiling time 30.714432299999316
compiled model time 18.933740727999975
```

### Versions

```
Collecting environment information...
PyTorch version: 2.0.1+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Debian GNU/Linux 10 (buster) (x86_64)
GCC version: (Debian 8.3.0-6) 8.3.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.28

Python version: 3.10.0 (default, Mar  3 2022, 09:58:08) [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.15.0-1041-azure-x86_64-with-glibc2.28
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
Address sizes:       48 bits physical, 48 bits virtual
CPU(s):              16
On-line CPU(s) list: 0-15
Thread(s) per core:  2
Core(s) per socket:  8
Socket(s):           1
NUMA node(s):        1
Vendor ID:           AuthenticAMD
CPU family:          25
Model:               1
Model name:          AMD EPYC 7763 64-Core Processor
Stepping:            1
CPU MHz:             2445.432
BogoMIPS:            4890.86
Virtualization:      AMD-V
Hypervisor vendor:   Microsoft
Virtualization type: full
L1d cache:           32K
L1i cache:           32K
L2 cache:            512K
L3 cache:            32768K
NUMA node0 CPU(s):   0-15
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl tsc_reliable nonstop_tsc cpuid extd_apicid aperfmperf pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy svm cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw topoext invpcid_single vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves clzero xsaveerptr rdpru arat npt nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold v_vmsave_vmload umip vaes vpclmulqdq rdpid fsrm

Versions of relevant libraries:
[pip3] flake8==6.0.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.24.3
[pip3] torch==2.0.1+cpu
[pip3] torch-geometric==2.3.1
[pip3] torch-scatter==2.1.1
[pip3] torchvision==0.15.2
[conda] numpy                     1.24.3                   pypi_0    pypi
[conda] torch-geometric           2.3.1                    pypi_0    pypi
[conda] torchvision               0.15.2                   pypi_0    pypi
```

cc @ezyang @msaroufim @wconstab @bdhirsh @anijain2305

**Comment by ZailiWang** [comment]
Hi @mattbarrett98 torch.compile is a JIT compilation approach for graph mode DAG generation, so it also requires "warmup" forward passes (like using torch.jit). Adding a for loop at the compiled model forward func, you can see the following forward passes (after 2nd) much faster.

**Comment by mattbarrett98** [comment]
> Hi @mattbarrett98 torch.compile is a JIT compilation approach for graph mode DAG generation, so it also requires "warmup" forward passes (like using torch.jit). Adding a for loop at the compiled model forward func, you can see the following forward passes (after 2nd) much faster.

ah I see, a bit lazy from me 😅 think I'm just used to jax's jit which only ever seems to need 1 "warmup" pass
